doc/api/http2.md

```diff
@@ -1474,6 +1474,12 @@ added: v8.4.0
     used to determine the padding. See [Using options.selectPadding][].
   * `settings` {[Settings Object][]} The initial settings to send to the
     remote peer upon connection.
+  * `Http2ServerRequest` {Http2ServerRequest} Specifies the ServerRequest class to
+    used. Useful for extending the original `Http2ServerRequest`.
+    Defaults to: `Http2ServerRequest`
+  * `Http2ServerRequest` {Http2ServerResponse} Specifies the ServerRequest class to
+    used. Useful for extending the original `Http2ServerResponse`.
+    Defaults to: `Http2ServerResponse`
```

Comment:
We would need to add the documentation of the constructor parameters. Those are not documented atm, but if we want to support this use case we should do it.